NDTensors/ext/NDTensorsMetalExt/adapt.jl

```diff
@@ -6,6 +8,8 @@ end
 
 # More general than the version in Metal.jl
 function Adapt.adapt_storage(arraytype::Type{<:MtlArray}, xs::AbstractArray)
-  arraytype_specified = specify_parameters(arraytype, get_parameters(xs))
+  arraytype_specified = specify_parameters(
+    arraytype, parameter_names(arraytype), parameters(xs)
+  )
```

Comment:
I'm still confused by this function call. What if `parameter_names(arraytype)` doesn't correspond with the output of `parameters(xs)`? For example, `xs` could just have an element type parameter, in which case it just outputs something like `(Float64,)`, but `parameter_names(arraytype)` outputs `(eltype, ndims, storagemode)`.

It seems like this should be:
```julia
  arraytype_specified = specify_parameters(
    arraytype, parameter_names(xs), parameters(xs)
  )
```
instead.